src/representation_theory/monoidal_closed.lean

```diff
@@ -0,0 +1,292 @@
+/-
+Copyright (c) 2022 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import representation_theory.Action
+
+/-!
+# `Action V G` is a closed monoidal category when `G : Group` and `V` is a closed monoidal category.
+-/
+
+universe u
+
+open opposite
+open category_theory
+
+variables
+  {V : Type (u+1)} [large_category V] [monoidal_category V] [monoidal_closed V] {G : Group.{u}}
+  (X Y Z : Action V G) (g g' : G)
+
+namespace Action
+
+/-- For f : X.V ⟶ Y.V, compose on the right by X.ρ g⁻¹ : X.V ⟶ X.V -/
+def pre_comp : (ihom X.V).obj Y.V ⟶ (ihom X.V).obj Y.V :=
+  (monoidal_closed.internal_hom.map (X.ρ (g⁻¹ : G)).op).app Y.V
+
+@[simp] lemma pre_comp_apply :
+  pre_comp X Y g = (monoidal_closed.internal_hom.map (X.ρ (g⁻¹ : G)).op).app Y.V := rfl
+
+@[simp] lemma pre_comp_mul :
+  pre_comp X Y (g * g') = pre_comp X Y g' ≫ pre_comp X Y g := by simp
+
+/-- For f : X.V ⟶ Y.V, compose on the left by Y.ρ g : Y.V ⟶ Y.V -/
+def post_comp : (ihom X.V).obj Y.V ⟶ (ihom X.V).obj Y.V :=
+  (monoidal_closed.internal_hom.flip.map (Y.ρ g)).app (op X.V)
+
+@[simp] lemma post_comp_apply :
+  post_comp X Y g = (monoidal_closed.internal_hom.flip.map (Y.ρ g)).app (op X.V) := rfl
+
+@[simp] lemma post_comp_mul :
+  post_comp X Y (g * g') = post_comp X Y g' ≫ post_comp X Y g := by simp
+
+lemma pre_post_comp_comm :
+  pre_comp X Y g ≫ post_comp X Y g' = post_comp X Y g' ≫ pre_comp X Y g := by simp
+
+/-- The internal hom of `X Y : Action V G`, i.e. the natural G-action on `[X, Y]` -/
+@[protected] def ihom (X Y : Action V G) : Action V G :=
+⟨(ihom X.V).obj Y.V,
+begin
+  refine ⟨λ g, pre_comp X Y g ≫ post_comp X Y g, _, _⟩,
+  { simp [pre_comp_apply, post_comp_apply],
+    apply category.id_comp },
```

Comment:
Mathlib's convention is to always avoid non-terminal simps (see https://leanprover-community.github.io/extras/simp.html#non-terminal-simps). You should either replace them by `simp only` or rearrange your proofs so that your `simp` always finishes the goal.